src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md

```diff
@@ -14,58 +14,37 @@ title: Storage Power Actor
-
-An invariant of the storage power consensus subsystem is that all storage in the power table must be verified. That is, miners can only derive power from storage they have already proven to the network.
-
-In order to achieve this, Filecoin delays updating power for new sector commitments until the first valid PoSt in the next proving period corresponding to that sector. (TODO: potential delay this further in order to ensure that any power cut goes undetected at most as long as the shortest power delay on new sector commitments).
-
-For instance, say a miner X does the following:
-- In epoch 100: commits 10 TB
-- In epoch 110: publishes a PoSt for their storage
-- In epoch 120: commits another 10TB
-- In epoch 135: publishes a new PoSt for their storage
+SPC provides a power table abstraction which tracks miner power (i.e. miner storage in relation to network storage) over time. The power table is updated for new sector commitments (incrementing miner power), for failed PoSts (decrementing miner power) or for other storage and consensus faults. `_updatePowerEntriesFromClaimedPower` is called to update a particular miner's entry in the power table when its claimed power has changed.
 
-Querying the power table for this miner at different rounds should yield (using the following shorthand as an illustration only):
-- `Power(X, 90) == 0`
-- `Power(X, 100) == 0`
-- `Power(X, 110) == 0`
-- `Power(X, 111) == 10`
-- `Power(X, 120) == 10`
-- `Power(X, 135) == 10`
-- `Power(x, 136) == 20`
-
-Conversely, storage faults only lead to power loss once they are detected (up to one proving period after the fault) so miners will mine with no more power than they have used to store data over time.
-
-Put another way, power accounting in the SPC is delayed between storage being proven or faulted, and power being updated in the power table (and so for leader election). This ensures fairness over time.
+Sector ProveCommit is the first time power is proven to the network and hence power is first added through `_rtAddPowerForSector` at `OnSectorProveCommit`. Power is also added when a sector's TemporaryFault period has ended. Miners are expected to prove over all their sectors that contribute to their power. `_rtDeductClaimedPowerForSectorAssert` is called to decrement a miner's power. This is called when a sector expires or invoked by miner through `OnSectorTerminate` and when a sector enters TemporaryFault through `OnSectorTemporaryFaultEffectiveBegin`. Both `_rtAddPowerForSector` and `_rtDeductClaimedPowerForSectorAssert` are currently called at `OnSectorModifyWeightDesc` as power is determined by `SectorStorageWeightDesc` and `SectorStorageWeightDesc` is only modified when a miner calls `ExtendSectorExpiration` to extend a sector's duration. This may or may not have an impact on power but the machinery is in place to preserve the flexibility.
```

Comment:
Nit: referencing internal method names here is extremely likely to rot. Better to keep the description conceptual.